src/model/song.rs

```diff
@@ -0,0 +1,53 @@
+use crate::Chord;
+use combine::{parser::char::spaces, sep_by, Parser, Stream};
+use std::fmt;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Song {
+    pub chords: Vec<Chord>,
+}
+
+impl Song {
+    pub fn new(chords: &[Chord]) -> Self {
+        Self {
+            chords: chords.to_vec(),
+        }
+    }
+
+    pub fn parser<Input>() -> impl Parser<Input, Output = Self>
+    where
+        Input: Stream<Token = char>,
+    {
+        sep_by(Chord::parser(), spaces()).map(|chords: Vec<_>| Self::new(&chords))
+    }
+}
+
+impl fmt::Display for Song {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let string = self
+            .chords
+            .iter()
+            .map(|c| c.to_string())
+            .collect::<Vec<_>>()
```

Comment:
Seems weird that you have to `.collect()` into a `Vec` before `.join()`'ing (vs just `.join()`'ing on an iterator directly)?

Reply:
Yeah, I don't understand why this is a function on `Slice` rather than `Iterator`.